components/canvas/webgl_thread.rs

```diff
@@ -1304,11 +1321,37 @@ impl WebGLImpl {
             WebGLCommand::Finish(ref sender) => Self::finish(ctx.gl(), sender),
             WebGLCommand::Flush => ctx.gl().flush(),
             WebGLCommand::GenerateMipmap(target) => ctx.gl().generate_mipmap(target),
-            WebGLCommand::CreateVertexArray(ref chan) => Self::create_vertex_array(ctx.gl(), chan),
-            WebGLCommand::DeleteVertexArray(id) => ctx.gl().delete_vertex_arrays(&[id.get()]),
-            WebGLCommand::BindVertexArray(id) => ctx
-                .gl()
-                .bind_vertex_array(id.map_or(0, WebGLVertexArrayId::get)),
+            WebGLCommand::CreateVertexArray(ref chan) => {
+                Self::create_vertex_array(ctx.gl(), use_apple_vertex_array, chan)
+            },
+            WebGLCommand::DeleteVertexArray(id) => {
+                let gl = ctx.gl();
+                let ids = [id.get()];
+                if use_apple_vertex_array {
```

Comment:
So are we just doing this stuff here because it's too inconvenient to do it in gleam?

Reply:
s/gleam/sparkle/, and we're doing this here because the GL type abstracts over operations that are common to both GL and GLES. That code does not currently have the information required to decide whether to use these apple methods or not.